Automattic/wp-calypso
# Account Recovery: Chain step4 -> step5 & 5b

## Summary
This PR chains the step 4 -> step 5 & 5b in the AR flow:
![demo-ar-4-5](https://cloud.githubusercontent.com/assets/1842898/25368308/2620e108-29ae-11e7-864d-4502c9306f40.gif)

In general, it consists of the following pieces:

1. Move `reset-password-sms-form` and `reset-password-email-form` directory one level up.
1. Chain the step 4 to step 5 & 5b by inspecting if the `method` field is set in the redux state.
1. Rename `pickResetMethod` as `setResetMethod` for better naming consistency.
1. The data layer handler for `ACCOUNT_RECOVERY_RESET_REQUEST` now dispatches `setResetMethod` action.
1. Wrap the form components of `<ResetPasswordForm />` in a `<form>` tag.
1. Step 5 & 5b can now navigate back to step 4 by clicking "No SMS" or "Didn't receive it?" anchor.

## Test Plan
1. Open an incognito window and make sure you are logged out.
1. Visit http://calypso.localhost:3000/account-recovery
1. Use whatever test accounts and follow instructions to proceed. You should be able to arrive the "email is sent" form and "sms message is sent" form, and navigate back to the "How would you like to reset your password?" form without problems.
